aderyn_core/src/context/browser/chain.rs

```diff
@@ -0,0 +1,639 @@
+use std::collections::HashMap;
+
+use crate::{
+    ast::*,
+    context::workspace_context::{ASTNode, WorkspaceContext},
+};
+
+/// GetParentChain allows us to grab the ancestral hirearchy of a given node in the AST
+/// all the way upto the ContractDefinition
+pub trait GetParentChain {
+    /// Get the parent Chain of an ASTNode
+    fn of(&self, context: &WorkspaceContext) -> Vec<ASTNode>;
+}
+
+impl GetParentChain for Assignment {
+    fn of(&self, context: &WorkspaceContext) -> Vec<ASTNode> {
+        get_parent_chain_of_child(self.id, context)
+    }
+}
+impl GetParentChain for BinaryOperation {
+    fn of(&self, context: &WorkspaceContext) -> Vec<ASTNode> {
+        get_parent_chain_of_child(self.id, context)
+    }
+}
+impl GetParentChain for Block {
+    fn of(&self, context: &WorkspaceContext) -> Vec<ASTNode> {
+        get_parent_chain_of_child(self.id, context)
+    }
+}
+impl GetParentChain for Conditional {
+    fn of(&self, context: &WorkspaceContext) -> Vec<ASTNode> {
+        get_parent_chain_of_child(self.id, context)
+    }
+}
+impl GetParentChain for ContractDefinition {
```

Comment:
It would be cool to have `GetParent` that just gets the direct parent too.

Reply:
That is coming in next PR : ) https://github.com/Cyfrin/aderyn/pull/168